### ISSUE TITLE
Publish to Azure Functions incomplete

### ISSUE THREAD

**Issue by LangdalP** [JetBrains/azure-tools-for-intellij#253]
Hi.

I have created a hello world azure function using the Rider template. Using the "publish to azure" configuration does not work. The build log seems to indicate that everything works (see below). However, what happens is that it creates a web app/azure functions with no actual functions. However, if I instead use the azure functions cli to run `func azure functionapp publish <NameOfApp>`, the deploy succeeds. This seems to indicate that the Rider template mostly works, but something is not quite right with the publish configuration/build config.

Rider version: 2019.2.2
Azure toolkit for Rider version: 3.26.0.534-2019.2
OS: Windows 10

*Edit:* Let me know if you need more diagnostics/debugging from me.

Build output:
```
Got existing Function App with Id: '/subscriptions/REDACTED/resourceGroups/bakst/providers/Microsoft.Web/sites/redacted'
Collecting 'Redacted' project artifacts...
Creating 'Redacted' project ZIP...
Project ZIP is created: 'C:\Users\redacted\AppData\Local\Temp\rider-publish393701045711205748.zip'
Stop Function App 'redacted'...
Publishing ZIP file. Attempt 1 of 3...
Published ZIP file successfully
Deleting ZIP file 'C:\Users\redacted\AppData\Local\Temp\rider-publish393701045711205748.zip'
Deploy succeeded.
Start Function App 'redacted'...
URL: https://redacted.azurewebsites.net
Done.
```

Here is the generated csproj-configuration for the functions project:
```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFramework>netcoreapp2.1</TargetFramework>
        <AzureFunctionsVersion></AzureFunctionsVersion>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.29" />
    </ItemGroup>
    <ItemGroup>
        <None Update="host.json">
            <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
        </None>
        <None Update="local.settings.json">
            <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
            <CopyToPublishDirectory>Never</CopyToPublishDirectory>
        </None>
    </ItemGroup>
</Project>
```

**Comment by maartenba** [comment]
Could you check a couple of things after publishing from Rider?

1) Under Advanced tools (Kudu) or App Service Editor in the Azure portal, do you see any directory named after your function containing a function.json file?
2) If yes, that means publish did work. If not, could you check the ZIP file that is being generated by Rider while publishing (will need to copy it while uploading, as Rider deletes it) and see if that contains your functions?

cc @sdubov who may also have some troubleshooting tips

**Comment by LangdalP** [comment]
@maartenba 
I did as you said, using Kudu.
The folder and file is created, but there is an error inside the function.json file. The problem is that the `scriptFile` element points to the temporary rider-publish folder. Specifically, the value of `scriptFile` is `../../../../../../pederlangdal/AppData/Local/Temp/rider-publish5438353661562244586/bin/Bakstbot.dll`

When I instead publish using `func azure functionapp publish <NameOfApp>`, the `scriptFile` element contains a value of `../bin/Bakstbot.dll`, which works.

**Comment by maartenba** [comment]
@sdubov could you have a look?

**Comment by maartenba** [comment]
@LangdalP is there any chance you could share the app's entire project folder with us? (without the .cs files/any secrets, but will need the entire project structure/.idea folder/...) - I'm maarten AT jetbrains com.

**Comment by LangdalP** [comment]
@maartenba 
I have now sent you an example of a solution where I have this problem, and also the build output produced when publishing to Azure.

Note that the solution is basically the simplest "Hello, World" Function App that is possible to make. I have just used the Rider template and added one HttpTrigger. Also note that I am using Windows 10.

**Comment by maartenba** [comment]
Thanks! Will investigate

**Comment by maartenba** [comment]
Thanks! Have checked, and indeed nothing "funky" to spot there. It seems that somehow when we invoke `msbuild /t:Publish`, an incorrect path is set in some case.

Do you have this issue on every publish or sporadically?

**Comment by LangdalP** [comment]
@maartenba For me, this happens on every publish with Rider.

**Comment by LangdalP** [comment]
On a side note: Publishing regular Web Apps works fine.

**Comment by maartenba** [comment]
The issue seems to have been fixed in the Azure Functions SDK:
https://github.com/Azure/azure-functions-vs-build-sdk/commit/1922f7d495a8089d8f3293e31dfbca1786e887b0

This is available/fixed since `Microsoft.NET.Sdk.Functions` version `1.0.30-beta4` - https://www.nuget.org/packages/Microsoft.NET.Sdk.Functions/1.0.30-beta4 - we recommend updating the `Microsoft.NET.Sdk.Functions` package to `1.0.30-beta4`.

**Comment by Robar666** [comment]
Hi,

it seems that I've encountered the exact same bug today. Is just created a plain Azure Function App in our subscription and  then created a solution with the plain HTTP trigger template. 

When I tried to upload it to Azure with Rider, I've received the exact [same output](https://github.com/JetBrains/azure-tools-for-intellij/issues/253#issue-506368374) (`Deploy succeeded.`), but (in contrast to the OP) the files weren't uploaded.

Workaround for me: I've tried it in Visual Studio and the initial upload worked. From thereon, the following deployments also worked with Rider.

@maartenba : I'm unsure if I should create a new question or if this comment here is enough.
